src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -44,6 +85,29 @@ protected override void HandleHeartbeatMessage(TransactionOperationContext jsonO
             DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Shiran, DevelopmentHelper.Severity.Normal, "Find a way to implement change vector updates for tackling optimization issues");
         }
 
+        protected override DynamicJsonValue GetHeartbeatStatusMessage(TransactionOperationContext context, long lastDocumentEtag, string handledMessageType)
+        {
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Shiran, DevelopmentHelper.Severity.Normal, "implement this");
+            string mergedChangeVector = "";
+            long lastEtag = int.MaxValue;
+            foreach (var handler in _handlers)
+            {
+                var lastChangeVector = handler.LastDatabaseChangeVector;
+                mergedChangeVector = ChangeVectorUtils.MergeVectors(mergedChangeVector, lastChangeVector);
+                var myEtag = handler.CurrentEtag;
+                lastEtag = Math.Min(lastEtag, myEtag);
+            }
+
+            LastReplicateChangeVector = mergedChangeVector;
+            LastEtag = lastEtag;
+
+            var heartbeat = base.GetHeartbeatStatusMessage(context, lastDocumentEtag, handledMessageType);
+            heartbeat[nameof(ReplicationMessageReply.CurrentEtag)] = LastEtag;
+            heartbeat[nameof(ReplicationMessageReply.DatabaseChangeVector)] = LastReplicateChangeVector;
+           
+            return heartbeat;
```

Comment:
we want to handle this in a different PR, right?

Reply:
Yep. Done